linera-service/src/persistent/mod.rs

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+mod file;
+
+use std::ops::{Deref, DerefMut};
+
+pub use file::File;
+
+pub trait Persistent: Deref {
```

Comment:
We could use some short (oneline) comment on public APIs.

Reply:
`PersistentValue` ?

Reply:
Agree on the docs, good catch :)  Though technically nothing in `linera-service` is ‘public’ as it can't be imported as a library, that will change in a little bit when this goes into `linera-client` :)

On the name — there's no clear consensus on this in Rust, but I prefer adjectives for trait names rather than nouns (as it avoids name clash with implementers, especially generic implementers).  The competing convention is verbs, so this could be named `Persist`; happy to switch to that if we think it's better.  If we want a noun then it should be agentive/patientive, i.e. `Persistable`, but suffixes like `-able` are specifically recommended against by the guidelines.

Reply:
In fact I think I'm convinced that `Persist` is more in line with Rust (`Read`, `Write`, et cetera) — I'll rewrite to that unless you have objections.

Reply:
I've added the docs and renamed `Persistent` to `Persist`.

Reply:
> I think Value is an essentially meaningless type name — anything nameable by a type is a value, so including the word ‘value’ in a type or trait name never adds information.

I meant `Value` for pure value or what people sometimes call "Pain Old Object" in OO languages.